Adding Camel Language Support for Emacs (#4)
camel-tooling/camel-lsp-client-emacs: README.md

> ## This is a client implementation of the Apache Camel Language Server Protocol for Emacs
> Link for Apache Camel Language Server 
> https://github.com/camel-tooling/camel-language-server/
> 
> # Text Editing capabilities of Camel URI with Camel XML DSL
> 
> Code completion for Camel LS with XML LS

Reviewer: ```suggestion
## This is a client implementation of the Apache Camel Language Server Protocol for Emacs
Link for Apache Camel Language Server 

# Text Editing capabilities of Camel URI with Camel XML DSL

For instance, code completion for Camel XML Dsl. The capabilities are based on the [Camel Language Server](https://github.com/camel-tooling/camel-language-server/).
```

![Demo](images/camell.gif)

NOTE: Completion is with Ctrl+Alt+i 

> ## prerequisites

Reviewer: ```suggestion
# Prerequisites
```

Java 11+ is required to be present on the System Path

> ## How to configure Emacs

Reviewer: ```suggestion
# How to configure Emacs
```

			:store-path lsp-camel-jar-file))

(defcustom lsp-camel-server-command `("java" "-jar" , lsp-camel-jar-file)
 "Camel server command."
 :type '(repeat string)
 :group 'lsp-camel
 :package-version '(lsp-mode . "8.0.0"))

(defun lsp-camel--create-connection ()
 (lsp-stdio-connection
  (lambda () lsp-camel-server-command)
  (lambda () (f-exists? lsp-camel-jar-file))))

(lsp-register-client
(make-lsp-client :new-connection (lsp-camel--create-connection)
				 :activation-fn (lsp-activate-on "xml")
				 :priority 0
				 :server-id 'camells
				 :add-on? t
				 :multi-root t
				 :initialized-fn (lambda (workspace)
								   (with-lsp-workspace workspace
									 (lsp--set-configuration (lsp-configuration-section "camel"))))
				 :download-server-fn (lambda (_client callback error-callback _update?)
									   (lsp-package-ensure 'camells callback error-callback))))

(lsp-consistency-check lsp-camel)

(provide 'lsp-camel)
;;; lsp-camel.el ends here
```
Opening the `lsp-camel.el` file

Evaluating it by calling `M-x eval-buffer`

Loading Camel LS by calling: `M-x lsp-install-server` and choosing `camells`

Opening a `camel.xml` file with this kind of content:

> ```lisp

Reviewer: ```suggestion
```xml
```

this one is xml code, not lisp

<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
	   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	   xsi:schemaLocation="
	   http://www.springframework.org/schema/beans https://www.springframework.org/schema/beans/spring-beans.xsd
	   http://camel.apache.org/schema/spring https://camel.apache.org/schema/spring/camel-spring.xsd">

  <camelContext id="camel" xmlns="http://camel.apache.org/schema/spring">
	<route id="a route">
	  <from uri="timer:timerName?delay=1000&amp;exchangePattern=InvalidEnumValue"/>
	  <to uri="direct:drink"/>
	</route>
  </camelContext>
</beans>

> ```

Reviewer: Would be nice to add something like:
Then you can enjoy editing capabilities of Camel URI. For instance by calling completion with Ctrl+Alt+i in the uri attribute value.
